File: app/domain/models/session.py
from __future__ import annotations
from dataclasses import dataclass, field
from datetime import datetime, timezone, timedelta
from enum import Enum
from typing import Optional
import uuid


def _utcnow() -> datetime:
    return datetime.now(timezone.utc).replace(tzinfo=None)


class SessionStatus(str, Enum):
    PENDING = "PENDING"
    PENDING_MIKROTIK = "PENDING_MIKROTIK"
    ACTIVE = "ACTIVE"
    EXPIRED = "EXPIRED"
    REVOKED = "REVOKED"


@dataclass
class HotspotSession:
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    session_id: str = field(default_factory=lambda: str(uuid.uuid4()))
    hotspot_id: str = ""
    plan_id: Optional[str] = None
    payment_id: Optional[str] = None
    client_phone: Optional[str] = None
    client_mac: Optional[str] = None
    mikrotik_username: str = ""
    mikrotik_password: str = ""
    profile: str = "default"
    status: SessionStatus = SessionStatus.PENDING
    bytes_in: int = 0
    bytes_out: int = 0
    activated_at: Optional[datetime] = None
    expires_at: Optional[datetime] = None
    created_at: datetime = field(default_factory=_utcnow)
    updated_at: datetime = field(default_factory=_utcnow)

    def activate(self) -> None:
        self.status = SessionStatus.ACTIVE
        self.activated_at = _utcnow()
        self.updated_at = self.activated_at

    def mark_expired(self) -> None:
        self.status = SessionStatus.EXPIRED
        self.updated_at = _utcnow()

    def revoke(self) -> None:
        self.status = SessionStatus.REVOKED
        self.updated_at = _utcnow()

File: Backend/Java/hotspot_pay/src/main/java/com/hotspotpay/session/model/Session.java
package com.hotspotpay.session.model;

import com.hotspotpay.session.enumeration.SessionStatus;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "sessions")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter @Setter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@EntityListeners(AuditingEntityListener.class)
public class Session {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @EqualsAndHashCode.Include
    @Column(updatable = false, nullable = false)
    private UUID id;

    @Column(name = "session_id", unique = true, nullable = false)
    private String sessionId;

    @Column(name = "payment_id", unique = true, nullable = true)
    private String paymentId;

    @Column(name = "hotspot_id", nullable = false)
    private String hotspotId;

    @Column(name = "plan_id", nullable = false)
    private String planId;

    @Column(name = "client_phone", nullable = false, length = 20)
    private String clientPhone;

    @Column(name = "client_mac", nullable = true, length = 17)
    private String clientMac;

    // Compte créé sur MikroTik
    @Column(name = "mikrotik_username", nullable = false, length = 100)
    private String mikrotikUsername;

    @Column(name = "mikrotik_password", nullable = false, length = 100)
    private String mikrotikPassword;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private SessionStatus status = SessionStatus.PENDING_MIKROTIK;

    @Column(name = "activated_at", nullable = false)
    private LocalDateTime activatedAt;

    @Column(name = "expires_at", nullable = false)
    private LocalDateTime expiresAt;

    @Column(name = "expired_at")
    private LocalDateTime expiredAt;

    @Column(name = "bytes_in")
    @Builder.Default
    private Long bytesIn = 0L;

    @Column(name = "bytes_out")
    @Builder.Default
    private Long bytesOut = 0L;

    @CreatedDate
    @Column(name = "created_at", updatable = false, nullable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}